### PR TITLE
spidershim: implement out v8::Isolate::<Add,Remove>GC<Epilogue,Prologue>Ca…

### PR DESCRIPTION
**Pull request by brendandahl**
follow up in #107

### CODE REVIEW

File: deps/spidershim/src/v8isolate.cc [hunk 1]
```diff
@@ -193,6 +193,22 @@ void Isolate::AddStackTrace(StackTrace* trace) {
   pimpl_->stackTraces.push_back(trace);
 }
 
+void Isolate::AddGCPrologueCallback(
+  GCCallback callback, GCType gc_type_filter) {
+  // TODO: used only with dtrace/etw, could be implemented with JS_SetGCCallback
```
Comment (ehsan) [inline deps/spidershim/src/v8isolate.cc]: @brendandahl given that we already have `OnGC()` in `v8isolate.cc`, any reason why we wouldn't just hook these up for reals? :)

**Comment by ehsan** [comment]
Ping?

File: deps/spidershim/src/v8isolate.cc (hunk 1 above)
Comment (brendandahl) [inline deps/spidershim/src/v8isolate.cc]: > @brendandahl given that we already have OnGC() in v8isolate.cc, any reason why we wouldn't just hook these up for reals? :)

I'll look into hooking them up now. I've been mainly stubbing out what chakra got away with.

**Comment by brendandahl** [comment]
@ehsan ready for review

**Comment by ehsan** [comment]
Looks great, thanks!

**Pull request merged**